pkg/controller/webhook/webhook_certificates_controller.go

```diff
@@ -92,10 +92,10 @@ func newReconciler(mgr manager.Manager, webhookParams Params, clientset kubernet
 }
 
 // Add adds a new Controller to mgr with r as the reconcile.Reconciler
-func Add(mgr manager.Manager, webhookParams Params, clientset kubernetes.Interface, webhook AdmissionControllerInterface) error {
+func Add(mgr manager.Manager, webhookParams Params, clientset kubernetes.Interface, webhook AdmissionControllerInterface, operatorParams operator.Parameters) error {
 	r := newReconciler(mgr, webhookParams, clientset)
 	// Create a new controller
-	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: r})
```

Comment:
This is the primary thing I'm concerned about in these new changes.  Was there a specific reason why the webhook controller wasn't using the common `NewController` func?

Reply:
I _think_ it's because we don't want/need to tune `MaxConcurrentReconciles` in the context of the webhook certificates.

Maybe we could introduce a function that accepts a predefined set of `controller.Options` to get the same behaviour: 

```golang
// NewController creates a new controller with the given name, reconciler and parameters and registers it with the manager.
func NewController(mgr manager.Manager, name string, r reconcile.Reconciler, p operator.Parameters) (controller.Controller, error) {
	options := controller.Options{Reconciler: r, MaxConcurrentReconciles: p.MaxConcurrentReconciles}
	return NewControllerWithOptions(mgr, name, p, options)
}

// NewControllerWithOptions creates a new controller with the given name, reconciler, parameters, options and registers it with the manager.
func NewControllerWithOptions(mgr manager.Manager, name string, p operator.Parameters, options controller.Options) (controller.Controller, error) {
	c, err := controller.New(name, mgr, options)
	if err != nil {
		return nil, err
	}
	return newNamespaceAwareWatchersController(c, p.ManagedNamespaces, p.OperatorNamespace), nil
}
``` 

And then use it here (`webhook_certificates_controller.go`):

```golang
// Add adds a new Controller to mgr with r as the reconcile.Reconciler
func Add(mgr manager.Manager, webhookParams Params, clientset kubernetes.Interface, webhook AdmissionControllerInterface, operatorParams operator.Parameters) error {
	r := newReconciler(mgr, webhookParams, clientset)
	// Create a new controller
	c, err := common.NewControllerWithOptions(mgr, ControllerName, operatorParams, controller.Options{Reconciler: r})
	if err != nil {
		return err
	}

	secret := types.NamespacedName{
		Namespace: webhookParams.Namespace,
		Name:      webhookParams.SecretName,
	}
...
```

Reply:
this seems sensible now that you've noted the potential reasons, and I've implemented this.